### PR TITLE
fix(client): avoid treating numeric LaTeX as currency in preprocessLaTeX

### PR DESCRIPTION
**Pull request by lesj0610**
## Summary
Fixes a LaTeX preprocessing bug where numeric-leading inline math was misclassified as currency and escaped, which broke KaTeX rendering.

Example that currently breaks:
- `Multiply: $65 \\times 44$`

Current broken transformation:
- `Multiply: \\$65 \\times 44$`

Expected transformation:
- `Multiply: $$65 \\times 44$$`

## Root Cause
`preprocessLaTeX` runs currency escaping before single-dollar math conversion.
The currency regex matches `$65` at the start of `$65 \\times 44$`, so the opening `$` is escaped as currency even though it belongs to a math span.

## Changes
- In `client/src/utils/latex.ts`:
  - Added a heuristic guard to detect whether a currency-like opening `$` is actually the start of a math expression.
  - If it is math, currency escaping is skipped for that `$`.
  - Added helpers:
    - `findNextUnescapedSingleDollar`
    - `looksLikeMathExpression`
    - `isMathDollarOpening`
- In `client/src/utils/latex.spec.ts`:
  - Added regression tests for numeric-leading LaTeX:
    - `Multiply: $65 \\times 44$`
    - `Compute $65 + 44 = 109$`

## Notes
- Existing currency behavior remains unchanged for plain currency strings (e.g. `Price is $50 and $100`).

## Follow-up (Thinking/Reasoning Rendering)
A second fix is included in this PR:
- `client/src/components/Chat/Messages/Content/Parts/Thinking.tsx` now renders thinking content via `MarkdownLite` and applies `preprocessLaTeX` when LaTeX parsing is enabled.
- This fixes the case where final answers render math correctly but internal reasoning/thought blocks do not.

### CODE REVIEW

File: client/src/utils/latex.ts [hunk 1]
```diff
@@ -123,7 +190,7 @@ export function preprocessLaTeX(content: string): string {
 
   let match: RegExpExecArray | null;
   while ((match = CURRENCY_REGEX.exec(processed)) !== null) {
-    if (!isInCodeBlock(match.index, codeRegions)) {
+    if (!isInCodeBlock(match.index, codeRegions) && !isMathDollarOpening(processed, match.index)) {
```
[the remark this answers is not in the document]
Comment (lesj0610) [reply client/src/utils/latex.ts]: Good catch. I replaced per-match forward scanning with a single pass cache over single-dollar pairs (`buildMathOpeningCache`) and reused it during currency escaping, which removes the repeated linear scan pattern. Included in commit 88aa6fb52.

File: client/src/utils/latex.spec.ts [hunk 2]
```diff
@@ -202,6 +202,18 @@ y$ which spans lines`;
...
+    const expected = 'Multiply: $$65 \\times 44$$';
+    expect(preprocessLaTeX(content)).toBe(expected);
+  });
+
+  test('does not treat numeric-leading arithmetic LaTeX as currency', () => {
+    const content = 'Compute $65 + 44 = 109$';
+    const expected = 'Compute $$65 + 44 = 109$$';
+    expect(preprocessLaTeX(content)).toBe(expected);
+  });
+
```
[the remark this answers is not in the document]
Comment (lesj0610) [reply client/src/utils/latex.spec.ts]: Added the suggested edge-case tests and also covered numeric-leading symbolic math terms (`$2n$`, `$3x$`) so they are treated as LaTeX math instead of currency. Included in commit 88aa6fb52.

**Comment by lesj0610** [comment]
Follow-up: I replied inline to both Copilot review threads and pushed additional changes in commit `88aa6fb52`.

Inline replies:
- https://github.com/danny-avila/LibreChat/pull/11899#discussion_r2839629235
- https://github.com/danny-avila/LibreChat/pull/11899#discussion_r2839629237

If it helps, I can also push a focused fix addressing the maintainer notes about `LATEX_OPERATOR_REGEX` scope and the `Thinking.tsx` whitespace behavior.
